src/ucp/rndv/rndv_ppln.c

```diff
@@ -84,43 +85,51 @@ ucp_proto_rndv_ppln_probe(const ucp_proto_init_params_t *init_params)
         return;
     }
 
-    /* Find the performance range of sending one fragment */
-    if (!ucp_proto_select_get_valid_range(select_elem->thresholds,
-                                          &frag_min_length, &frag_max_length)) {
-        return;
-    }
+    /* Add each proto as a separate variant */
+    ucs_array_for_each(proto, &select_elem->proto_init.protocols) {
+        /* Skip empty variants and reconfig proto */
+        if ((proto->caps.num_ranges == 0) || (proto->proto_id == 0)) {
```

Comment:
I added assert for `proto->caps.num_ranges > 0` but left reconfig proto skip since this proto has infinite latency. If reconfig proto passed forward as other protocols do, then some arithmetic operations will happen with infinite latency and that causes weird failures.